src/gui/PropertyModel.cpp

```diff
@@ -0,0 +1,64 @@
+#include <PropertyModel.h>
+#include <iostream>
+PropertyModel::PropertyModel(QObject* parent) : QAbstractTableModel(parent)
+{
+
+}
+
+PropertyModel::~PropertyModel()
+{
+
+}
+
+int PropertyModel::rowCount(const QModelIndex& ) const
+{
+    return properties.size();
+}
+
+int PropertyModel::columnCount(const QModelIndex& ) const
+{
+    return 2;
+}
+
+QVariant PropertyModel::data(const QModelIndex& index, int role) const
+{
+    if (role == Qt::DisplayRole)
+    {
+        if (index.column() == 0)
+            return properties[index.row()];
+        else if (index.column() == 1)
+            return propertySource->property(properties[index.row()].toStdString().c_str());
+    }
```

Comment:
I would add at least a comment for the else-path, or log an error.

Reply:
This code path should never happen (unless there is a bug in Qt) since columnCount is always 2, so it didn't look as worth handling, but I can change that of course

Reply:
If it is impossible, then I suggest
```c++
else { assert(false); /* BUG! */ }
```

Reply:
In my opinion this is not correct way to handle the situation. To cite the QAbstractItemModel subclassing documentation:
Note: If you do not have a value to return, return an invalid QVariant instead of returning 0.
http://doc.qt.io/qt-5/qabstractitemmodel.html#data
I don't think this is a situation that should be handled in our code at all.

Reply:
1. Enclose if body with braces (here and in every other statements). The code may be modified later by less experienced engineers and not having braces is error-prone.
2. Else would be nice here indeed. It will probably never happen, but if it will we will be sure that the program is in the deep shit ;-)

Reply:
@tuvok I proposed a no-return case that will exist only in the debug mode, mainly to help debugging. In the production mode, it leads to a return of an invalid variant (even if the else-path is taken).